dds/DCPS/DataReaderImpl.cpp

```diff
@@ -2248,9 +2248,8 @@ DataReaderImpl::instances_liveliness_update(WriterInfo& info,
     const ACE_Time_Value& when)
 {
   ACE_GUARD(ACE_Recursive_Thread_Mutex, instance_guard, this->instances_lock_);
-  for (SubscriptionInstanceMapType::iterator iter = instances_.begin(),
-      next = iter; iter != instances_.end(); iter = next) {
-    ++next;
+  for (SubscriptionInstanceMapType::iterator iter = instances_.begin();
+       iter != instances_.end(); ++iter) {
```

Comment:
I made this change but I should've have asked why it has the `next` iterator, because it looks redundant to me.

Reply:
Looks like it would be used if the call to writer_become_dead invalidated the iterator (even indirectly).